### DISCUSSION TITLE
LoadDocument at reduce action in Map-Reduce Index

### DISCUSSION THREAD

**Discussion by panagushin** [ravendb/ravendb#14603] [Q&A] [answered]
It is possible to LoadDocument during Reduce action?
I am trying to load and getting this error:
System.ArgumentException: **Cannot execute Load. Source does not have a key.**
But I know that document I'm looking for is exist and I'm passing it's Id to LoadDocument function

**Comment by ayende** [comment, marked as answer]
No, the `reduce` stage isn't valid for that.
You have to load the document from the `map` stage.

That said, what are you trying to do?

**Reply by panagushin** [reply]
I solved my problem by creating another index on top of this map-reduce, and load document at it's map action.
RavenDB is awesome :)
